### DISCUSSION TITLE
A new way to managing dependencies

### DISCUSSION THREAD

**Discussion by idranme** [denoland/deno#19543] [Ideas]
Reference: https://github.com/denoland/experimental-deno-specifiers-example/blob/main/02-deno-canonical-app/

`project1/import_map.json`:
```json
{
    "imports": {
        "colors": "https://deno.land/std@0.192.0/fmt/colors.ts",
        "chalk": "npm:chalk@^1.0",
        "dax": "deno:dax@^23.0/mod.ts"
    }
}
```

`project1/export.ts`:
```ts
/** Feature: Specify the importmap used by the current project */
// @deno-export-deps ./import_map.js

/** Feature: Automatic detection of importmap within the project */
import { red } from "colors"

function output(){
    console.log(red("hello world"))
}

export {
    output
}
```

`project2/import.ts`:
```ts
/** Feature: Automatic detection of exported importmap */
import { output } from "../project1/export.ts"

output()
```

The purpose is to serve as an alternative to [`deps.ts`](https://deno.com/manual@v1.34.3/examples/manage_dependencies).

**Comment by jamesbradlee** [comment]
IIRC Deno didn't want to add a deno specifier for 3rd party modules because it breaks the purpose of url imports. Personally I prefer URL imports and dep.ts/managing import maps. There are 3rd party tools that can do this for you if you want to manage dependencies with a tool.

Personally I find it quicker to use deps.ts + url import + intellisense
